src/librustc_platform_intrinsics/lib.rs

```diff
@@ -37,60 +36,41 @@ pub enum IntrinsicDef {
     Named(&'static str),
 }
 
+
+static VOID: Type = Type::Void;
+
 static I8: Type = Type::Integer(true, 8, 8);
 static I16: Type = Type::Integer(true, 16, 16);
 static I32: Type = Type::Integer(true, 32, 32);
 static I64: Type = Type::Integer(true, 64, 64);
-static U8: Type = Type::Integer(false, 8, 8);
-static U16: Type = Type::Integer(false, 16, 16);
-static U32: Type = Type::Integer(false, 32, 32);
-static U64: Type = Type::Integer(false, 64, 64);
```

Comment:
> The LLVM intrinsic definitions themselves don't use unsigned values, all their type signatures are signed except for floating point values. Therefore the generation program is only generating signed intrinsics. 

It's not that the LLVM intrinsics have signed or unsigned integers in their signatures, LLVM integer types simply don't have a signedness, instead they put it in the operations. But that doesn't mean the Rust interface to those intrinsics should ignore Rust's richer type systems. Requiring wrapper crates to insert lots of ugly `transmute` would be a pointless complication IMHO, and would potentially generate worse code (in debug mode at least). To me this points to a weakness of the "generate everything from LLVM definitions" approach more than anything else.